google/cloud/storage/internal/curl_client.cc

```diff
@@ -1056,6 +1068,73 @@ CurlClient::WriteObjectXml(InsertObjectStreamingRequest const& request) {
       std::unique_ptr<internal::ObjectWriteStreambuf>(std::move(buf)));
 }
 
+std::pair<Status, ObjectMetadata> CurlClient::InsertObjectMediaMultipart(
+    InsertObjectMediaRequest const& request) {
+  CurlRequestBuilder builder(
+      upload_endpoint_ + "/b/" + request.bucket_name() + "/o", upload_factory_);
+  SetupBuilder(builder, request, "POST");
+  // Pick a separator that does not conflict with the request contents:
+  auto boundary = PickBoundary(request.contents());
+  builder.AddHeader("content-type: multipart/related; boundary=" + boundary);
+  builder.AddQueryParameter("uploadType", "multipart");
+  builder.AddQueryParameter("name", request.object_name());
+
+  std::unique_ptr<internal::CurlStreambuf> buf(
+      new internal::CurlStreambuf(builder.BuildUpload(),
+                                  client_options().upload_buffer_size()));
+  ObjectWriteStream writer(std::move(buf));
+
+  nl::json metadata = nl::json::object();
+  if (request.HasOption<MD5HashValue>()) {
+    metadata["md5Hash"] = request.GetOption<MD5HashValue>().value();
+  }
+
+  std::string marker = "--" + boundary + "\r\n";
```

Comment:
Nit: Be explicit about all the `"\r\n"` line endings (i.e. don't add them to this `marker`).  Then in the lines below it's a bit more obvious where the newlines are present, how many there are, etc.  It might also read more nicely if you stored those chars in their own string, e.g.:

```
std::string crlf = "\r\n";
writer << marker << crlf
       << "content-type: application/json; charset=UTF-8" << crlf
       << crlf
       << metadata.dump << crlf
       << marker << "--" << crlf;
```

Reply:
Done.